### PR TITLE
Feature/create

### PR DESCRIPTION
**Pull request by Shirley-Liz**
request a review

### CODE REVIEW

File: src/server/controllers/tasks.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import {Task} from '../models/tasks';
+import { Task } from "../models/tasks";
```
Comment (edwinmp) [inline src/server/controllers/tasks.js]: I think you have some formatter that's going against our `eslint` ... please turn it off.
No spaces between the curly braces and quotes must be single, not double.

File: src/server/controllers/tasks.js [hunk 2]
```diff
@@ -24,32 +25,63 @@ export class TaskManager {
...
   }
   /**
-  *
-  *
-  * @param {*} [option={}]
-  * @memberof TaskManager
-  */
-  findById(option={}) {
+   *
+   *
```
Comment (edwinmp) [inline src/server/controllers/tasks.js]: Add a description for this function.

File: src/server/controllers/tasks.js [hunk 3]
```diff
@@ -24,32 +25,63 @@ export class TaskManager {
...
+   * @param {*} [option={}]
```
Comment (edwinmp) [inline src/server/controllers/tasks.js]: Why is option an object when you're finding by ID?

Comment (edwinmp) [reply src/server/controllers/tasks.js]: Shouldn't be named option.

File: src/server/controllers/tasks.js [hunk 4]
```diff
@@ -24,32 +25,63 @@ export class TaskManager {
...
+   * @memberof TaskManager
+   */
+  findById(option = {}) {
     Task.findById(option, (error, tasks) => {
       if (error) {
         let response = error.message;
-        if (error.kind === 'ObjectId') {
-          response = 'Invalid Id';
+        if (error.kind === "ObjectId") {
```
Comment (edwinmp) [inline src/server/controllers/tasks.js]: Single quotes

File: src/server/controllers/tasks.js [hunk 5]
```diff
@@ -24,32 +25,63 @@ export class TaskManager {
...
-          error: response,
+          error: response
         });
       } else {
         this.res.json(tasks);
       }
     });
   }
+
+  insertTask() {
```
Comment (edwinmp) [inline src/server/controllers/tasks.js:100]: Add a JSDoc.

File: src/server/controllers/tasks.js [hunk 6]
```diff
@@ -24,32 +25,63 @@ export class TaskManager {
...
+    const newTask = new Task(this.req.body);
+    newTask.save((error, tasks) => {
+      console.log("error");
```
Comment (edwinmp) [inline src/server/controllers/tasks.js]: Don't think we need this.

File: src/server/controllers/tasks.js [hunk 7]
```diff
@@ -24,32 +25,63 @@ export class TaskManager {
...
+      if (error) {
+        console.log("error")
```
Comment (edwinmp) [inline src/server/controllers/tasks.js]: Or this.

File: src/server/controllers/tasks.js [hunk 8]
```diff
@@ -24,32 +25,63 @@ export class TaskManager {
...
+        let response = error.message;
```
Comment (edwinmp) [inline src/server/controllers/tasks.js]: Inline this.

File: src/server/controllers/tasks.js [hunk 9]
```diff
@@ -24,32 +25,63 @@ export class TaskManager {
...
+
+        this.res.json({
+          error: response
+        });
+      } else {
+        console.log("error")
```
Comment (edwinmp) [inline src/server/controllers/tasks.js]: Don't need this.

File: src/server/controllers/tasks.js [hunk 10]
```diff
@@ -24,32 +25,63 @@ export class TaskManager {
...
+        this.res.json(tasks);
+      }
+    });
+
+    // this.res.json(newTask);
```
Comment (edwinmp) [inline src/server/controllers/tasks.js]: Don't commit unnecessary comments.

File: src/server/models/tasks.js [hunk 11]
```diff
@@ -1,15 +1,33 @@
...
-  name: String,
-  body: String,
-  date: Date,
+  name: {
+    type: String,
+    required: true,
+  },
+  body:
+  {
+    type:String,
```
Comment (edwinmp) [inline src/server/models/tasks.js]: Space after `type:`

File: src/server/models/tasks.js [hunk 12]
```diff
@@ -1,15 +1,33 @@
...
+    required: true,
+  },
+  date:{
+    type:Date,
```
Comment (edwinmp) [inline src/server/models/tasks.js]: Space after `type:`

File: src/server/models/tasks.js [hunk 13]
```diff
@@ -1,15 +1,33 @@
 import * as mongoose from 'mongoose';
+import { toUnicode } from 'punycode';
```
Comment (edwinmp) [inline src/server/models/tasks.js]: What's the purpose of this?

File: src/server/models/tasks.js [hunk 14]
```diff
@@ -1,15 +1,33 @@
...
   id: ObjectId,
-  name: String,
-  body: String,
-  date: Date,
+  name: {
+    type: String,
+    required: true,
+  },
+  body:
+  {
```
Comment (edwinmp) [inline src/server/models/tasks.js]: Move to previous line.

File: src/server/models/tasks.js [hunk 15]
```diff
@@ -1,15 +1,33 @@
...
+    type:String,
+    required: true,
+  },
+  date:{
```
Comment (edwinmp) [inline src/server/models/tasks.js]: Space after `date:`

File: src/server/models/tasks.js [hunk 16]
```diff
@@ -1,15 +1,33 @@
...
+    type:Date,
+    required: true,
+  },
+  email:{
```
Comment (edwinmp) [inline src/server/models/tasks.js]: Space after `email:`

File: src/server/server.js [hunk 17]
```diff
@@ -24,5 +28,8 @@ app.get('/', (req, res) => res.send('Hello World!'));
 
 app.get('/api/tasks', (req, res) => new TaskManager(req, res).find());
 app.get('/api/tasks/:id', (req, res) => new TaskManager(req, res).findById(req.params.id));
+app.post('/api/task', (req, res) => new TaskManager(req, res).insertTask());
+
+
```
Comment (edwinmp) [inline src/server/server.js]: Remove extra blank lines.

**Review by edwinmp — CHANGES_REQUESTED** [review]

**Comment by Shirley-Liz** [comment]
changes made

File: .eslintrc.json [hunk 18]
```diff
@@ -15,6 +15,5 @@
         "sourceType": "module"
     },
     "rules": {
-        "max-len": [ "error", { "code": 100 } ]
```
Comment (edwinmp) [inline .eslintrc.json]: Why did you remove the rule? It served a purpose... revert

File: .eslintrc.json [hunk 19]
```diff
@@ -15,6 +15,5 @@
...
     }
-}
+}
```
Comment (edwinmp) [inline .eslintrc.json]: New line at the end of each file.

**Review by edwinmp — CHANGES_REQUESTED** [review]
There's some unresolved comments. Also, update your local `develop` branch, then switch back to this one and rebase `develop` ... i.e `git rebase develop` ... this will attempt to full the changes on the develop branch into yours ... you'll run into some conflicts... resolving them should be a good experience for you.

**Comment by Shirley-Liz** [comment]
made changes

File: src/server/controllers/tasks.js [hunk 20]
```diff
@@ -68,5 +74,21 @@ export class TaskManager {
       }
     });
   }
-}
 
+  insertTask() {
+    const newTask = new Task(this.req.body);
+    newTask.save((error, tasks) => {
+      console.log('error');
```
Comment (edwinmp) [inline src/server/controllers/tasks.js]: This console log serves no purpose.

File: src/server/controllers/tasks.js [hunk 21]
```diff
@@ -68,5 +74,21 @@ export class TaskManager {
...
+      if (error) {
+        console.log('error');
```
Comment (edwinmp) [inline src/server/controllers/tasks.js]: This one too...

File: src/server/controllers/tasks.js [hunk 22]
```diff
@@ -68,5 +74,21 @@ export class TaskManager {
...
+        const response = error.message;
+        this.res.json({
+          error: response,
+        });
+      } else {
+        console.log('error');
```
Comment (edwinmp) [inline src/server/controllers/tasks.js]: This one doesn't even make sense... there's no error here.

**Review by edwinmp — CHANGES_REQUESTED** [review]

File: src/server/controllers/tasks.js [hunk 23]
```diff
@@ -74,21 +74,10 @@ export class TaskManager {
...
-      if (error) {
-        console.log('error');
-        const response = error.message;
-        this.res.json({
-          error: response,
-        });
-      } else {
-        console.log('error');
-        this.res.json(tasks);
-      }
```
Comment (edwinmp) [inline src/server/controllers/tasks.js:113]: You deleted vital error handling... please add it again. My comment referred ONLY to the console logs.

**Review by edwinmp — CHANGES_REQUESTED** [review]

**Review by edwinmp — APPROVED** [review]

**Review by priscanicktar — APPROVED** [review]

**Pull request merged**